Week8/MovieViewer/app/src/main/java/com/wmw/movieviewer/repository/MovieRepository.kt

```diff
@@ -0,0 +1,39 @@
+package com.wmw.movieviewer.repository
+
+import androidx.lifecycle.LiveData
+import com.wmw.movieviewer.App
+import com.wmw.movieviewer.model.Movie
+import com.wmw.movieviewer.model.MovieDao
+import com.wmw.movieviewer.model.Success
+import com.wmw.movieviewer.networking.RemoteApi
+
+class MovieRepository(private val movieDao: MovieDao, private val movieApi: RemoteApi) {
+
+    fun getAllMovies(): LiveData<List<Movie>> = dao.getAllMoviesSortedByTitle()
+
+    suspend fun refreshMovies() {
+        val result = movieApi.getTopMovies()
+
+        if (result is Success) {
+            movieDao.insert(result.data.data.movies.toList().map {
+                Movie(
+                    it.idIMDB,
+                    it.releaseDate,
+                    it.title,
+                    it.plot,
+                    it.genres,
+                    it.urlPoster,
+                    it.ranking
+                )
+            })
+        } else {
+            print("Error: $result")
+        }
+    }
+
+    suspend fun deleteMovie(movie: Movie) = dao.deleteMovie(movie)
+
+    suspend fun getMovieById(movieId: String?) = dao.getMovieById(movieId)
+
+    private val dao = App.movieDb.movieDao()
```

Comment:
Not sure why this is at the bottom, properties and things you use throughout your class should be at the top.

Reply:
Also: Pass teh DAO in through the constructor

Reply:
The sorting pattern I have been using, puts all private variables, and methods at the bottom.
Sorting by visibility.